packages/api/tests/endpoints/test-executions.js

```diff
@@ -170,3 +370,226 @@ test('GET fails if execution is not found', async (t) => {
   t.is(response.status, 404);
   t.true(response.body.message.includes('No record found for'));
 });
+
+test('DELETE removes only specified execution from all data stores', async (t) => {
+  const { knex } = t.context;
+
+  const newExecution = fakeExecutionFactoryV2({
+    arn: 'arn3',
+    status: 'completed',
+    name: 'test_execution',
+  });
+
+  await executionModel.create(newExecution);
+  const executionPgRecord = await translateApiExecutionToPostgresExecution(
+    newExecution,
+    knex
+  );
+  await executionPgModel.create(knex, executionPgRecord);
+
+  t.true(await executionModel.exists({ arn: newExecution.arn }));
+  t.true(await executionPgModel.exists(knex, { arn: newExecution.arn }));
+
+  await request(app)
+    .delete(`/executions/${newExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  // Correct Dynamo and PG execution was deleted
+  t.false(await executionModel.exists({ arn: newExecution.arn }));
+
+  const dbRecords = await executionPgModel.search(t.context.knex, {
+    arn: newExecution.arn,
+  });
+
+  t.is(dbRecords.length, 0);
+
+  // Previously created executions still exist
+  t.true(await executionModel.exists({ arn: fakeExecutions[0].arn }));
+  t.true(await executionModel.exists({ arn: fakeExecutions[1].arn }));
+
+  const originalExecution1 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[0].arn,
+  });
+
+  t.is(originalExecution1.length, 1);
+
+  const originalExecution2 = await executionPgModel.search(t.context.knex, {
+    arn: fakeExecutions[1].arn,
+  });
+
+  t.is(originalExecution2.length, 1);
+});
+
+test('DELETE returns a 404 if Dynamo execution cannot be found', async (t) => {
+  const nonExistantExecution = {
+    arn: 'arn9',
+    status: 'completed',
+    name: 'test_execution',
+  };
+
+  const response = await request(app)
+    .delete(`/executions/${nonExistantExecution.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(404);
+
+  t.is(response.body.message, 'No record found');
+});
+
+test('POST /executions/search-by-granules returns correct executions when ids array is passed', async (t) => {
+  const { fakeGranules, fakePGExecutions } = t.context;
+
+  const response = await request(app)
+    .post('/executions/search-by-granules')
+    .send({ ids: [fakeGranules[0].granuleId, fakeGranules[1].granuleId] })
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`);
+
+  t.is(response.body.length, 3);
+
+  response.body.forEach((execution, index) => t.deepEqual(
```

Comment:
By using a `forEach` and assuming the index of the returned results matches the index of the order we created the PG records, we're assuming how the results are ordered when returned by the database. Obviously it's working for now, but it could change if we change how results are ordered in our query to search for records.

I think preferable solution is still to iterate but to find the record for comparison doing a `find` on `fakePgExecutions` where the `arn` values match and then comparing those records. This would apply throughout the tests

We don't need to change it for now, just a heads up.

Reply:
I was having some issues with the result order and this does future proof more than what I had, so I made that change